resources/qml/Cura.qml

```diff
@@ -45,7 +45,7 @@ UM.MainWindow
     WelcomeDialog
     {
         id: welcomeDialog
-        visible: false
+        visible: true  // True, so if somehow no preferences are found/loaded, it's shown anyway.
```

Comment:
Instead of setting both `visible` and `visibility` (via `show()` and `hide()`) I think it'll be easier to only use `visibility`. It needs to be set to `Hidden` when `needToShowUserAgreement` is false, or `Windowed` when it is true. Then you won't need to have the Component.onCompleted link any more down at line 77. So something like this:

`visibility: CuraApplication.needToShowUserAgreement ? Window.Windowed : Window.Hidden`

Reply:
After discussion: Decided to favour 'visible' instead of 'visibility/show/hide', becasue that didn't seem to work.